docs/src/pages/guide/how-to/00-llm-models.mdx

```diff
@@ -79,7 +79,63 @@ You can choose from several providers and models:
 
 - **Anthropic Vertex models**: 'claude-3-5-sonnet@20240620', 'claude-3-opus@20240229', 'claude-3-sonnet@20240229', 'claude-3-haiku@20240307'
 
-Mastra also supports other providers through Portkey and AI SDK. Here are [Portkey docs](https://docs.portkey.ai/docs/custom-models) and [an example](https://github.com/mastra-ai/mastra/blob/main/examples/vnext/src/mastra/agents/test.ts).
+## Custom Models
+
+While Mastra provides native support for popular LLM providers, you can also create custom model implementations for greater control and flexibility - even for supported providers. This allows you to fine-tune the integration to your specific needs.
+
+Here is an example of using a custom provider - Ollama.
+
+```bash npm2yarn copy
+npm install ollama-ai-provider
+```
+
+Import and configure the Ollama model by using `createOllama` from the `ollama-ai-provider` package.
+
+```typescript copy showLineNumbers
+import { createOllama } from "ollama-ai-provider";
+
+const ollama = createOllama({
+  // optional settings, e.g.
+  baseURL: "https://api.ollama.com",
+});
+```
+
+After creating the instance, you can use it like any other model in Mastra.
+
+```typescript copy showLineNumbers lines={9-14, 27} filename="src/mastra/index.ts"
+import { Mastra, type ModelConfig  } from "@mastra/core";
+import { createOllama } from "ollama-ai-provider";
+
+const ollama = createOllama({
+  // optional settings, e.g.
+  baseURL: "https://api.ollama.com",
+});
+
+const modelConfig: ModelConfig = {
+  model: ollama.chat("gemma");
+  apiKey: process.env.OLLAMA_API_KEY,
+  provider: "Ollama"; // Custom provider name
+  toolChoice: 'auto';
+};
+
+const mastra = new Mastra({});
+
+const llm = mastra.llm;
+
+const response = await llm.text({
+  messages: [
+    {
+      role: "user",
+      content: "What is machine learning?",
+    },
+  ],
+  model: modelConfig,
+});
+```
+
+If you were using an agent. The process is the same.
```

Comment:
The fact that we need to say this is a sign that this text is in the wrong place. We need to create a Model Config reference file under a new LLM folder and move a lot of text there. Start with this bit for this PR but you will want to do a follow up moving the Model lists there as well.

Reply:
Completely agree. Working on the new llm reference, and porting all of these over.

Reply:
We don't need to merge this.